sdk/scripts/check-crates.sh

```diff
@@ -25,6 +25,9 @@ printf "%s\n" "${files[@]}"
 
 has_error=0
 for file in "${files[@]}"; do
+  if [ -z "$file" ]; then
```

Comment:
how is an empty string getting in the array in the first place? implies broken logic elsewhere that should be fixed rather than monkey patched

Reply:
when `git diff "$COMMIT_RANGE" --diff-filter=AM --name-only | grep Cargo.toml | grep "^sdk/" ` returns nothing, the rest of the line 
` readarray -t files <<<"$(git diff "$COMMIT_RANGE" --diff-filter=AM --name-only | grep Cargo.toml | grep "^sdk/" | sed 's#sdk/##')" ` will still produce an array, albeit with empty strings. 

```
readarray -t bla <<< ""
echo ${#bla[@]} #prints 1
```

Reply:
CI should be unblocked first and foremost. I'll get a proper fix out

Reply:
ideally do that with revert, not hacks